src/BinSkim.Sdk/CompilerDataLogger.cs

```diff
@@ -20,14 +20,18 @@ public class CompilerDataLogger
         private const string CommandLineEventName = "CommandLineInformation";
         private const string AssemblyReferencesEventName = "AssemblyReferencesInformation";
 
-        private readonly int ChunkSize;
-        private readonly string sha256;
+        private string Sha256 => context?.Hashes?.Sha256;
+
+        private readonly int chunkSize;
         private readonly string relativeFilePath;
+        private readonly IAnalysisContext context;
+
+        internal static string s_sessionId;
 
-        private static string s_sessionId;
         private static bool s_printHeader = true;
```

Comment:
>private static bool s_printHeader = true; [](http://example.com/codeflow?start=8&length=41)

add private static ctor, all static initialization is explicit in a common place.